core/modules/wdb/QueryAction.cc

```diff
@@ -63,6 +63,8 @@
 #include "wdb/ChunkResource.h"
 #include "wdb/QueryAction.h"
 
+#define MAX_NAME_LENGTH 50
```

Comment:
I've already commented on using the POSIX constant HOST_NAME_MAX but just for information-transfer note that the actual value of that constant on Linux is typically 64, so this was not only an unnecessary constant definition but a bad value - and a good illustration of the need to follow POSIX more closely.  Please make sure that this idea is scrubbed from the rest of the code base.